platforms/core-runtime/stdlib-java-extensions/src/main/java/org/gradle/internal/service/scopes/Scope.java

```diff
@@ -17,75 +17,131 @@
+ * It can be a command-line invocation or a Tooling API client request (e.g. an IDE sync).
+ *
  * @see ServiceScope
  */
 public interface Scope {
 
     /**
-     * These services are reused across builds in the same process.
-     *
-     * <p>Global services are visible to all other services.</p>
+     * Scope of the entire <em>build process</em>, e.g. a long-lived Gradle daemon.
+     * <p>
+     * The build process can potentially serve multiple Gradle invocations.
+     * The <em>build process state</em> holds the global state of the build process and manages all the other state.
+     * <p>
+     * The build process state is managed by the {@code BuildProcessState} class.
+     * An instance is created once for a given process.
+     * <p>
+     * Global services are visible to all other services.
      */
     interface Global extends Scope {}
 
     /**
-     * These services are reused across builds in the same process while the Gradle user home directory remains unchanged.
-     * The services are closed when the Gradle user home directory changes.
-     *
-     * <p>{@link Global} and parent scope services are visible to {@link UserHome} scope services, but not vice versa.</p>
+     * The scope of a Gradle invocation particular Gradle user home directory.
+     * <p>
+     * When that directory changes between Gradle invocations, the state is discarded and recreated.
+     * Otherwise, the state is reused between invocations.
+     * <p>
+     * A regular Gradle invocation deals with a single user home directory,
+     * but the daemon can be potentially reused for invocations with other directories as well.
+     * <p>
+     * The related state is created per Gradle invocation.
+     * <p>
+     * {@link Global} services are visible to {@link UserHome} services and descendant scopes, but not vice versa.
      */
     interface UserHome extends Global {}
 
     /**
-     * These services are reused across build sessions.
+     * The scope of the state shared across {@link BuildSession build sessions}.
      * <p>
-     * Generally, one regular Gradle invocation is conceptually a session.
-     * However, the GradleBuild task is currently implemented in such a way that it uses a discrete session.
-     * Having the GradleBuild task reuse the outer session is complicated because it <a href="https://github.com/gradle/gradle/issues/4559">may use a different Gradle user home</a>.
+     * A regular Gradle invocation requires only one build session.
+     * However, when the {@code GradleBuild} task is involved, it can create "nested" build sessions.
+     * Having the {@code GradleBuild} task reuse the outer session is complicated because
+     * it <a href="https://github.com/gradle/gradle/issues/4559">can use a different Gradle user home</a>.
+     * <p>
+     * The cross build session state is managed by the {@code CrossBuildSessionState} class.
+     * An instance is created per Gradle invocation.
```

Comment:
💅 Perhaps we should also add that unlike `UserHome` and `Global` this scope is discarded at the end of the invocation.

Reply:
Added